lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -510,6 +510,24 @@ static bool isOkToBitSelectFrom(Value v) {
   return false;
 }
 
+/// Find a nested IfOp in an else block that can be printed as `else if`
+/// instead of nesting it into a new `begin` - `end` block.  The block must
+/// contain a single IfOp and optionally expressions which can be hoisted out.
+static IfOp findNestedElseIf(Block *elseBlock) {
+  IfOp ifOp;
+  for (auto &op : *elseBlock) {
+    if (auto opIf = dyn_cast<IfOp>(op)) {
+      if (ifOp)
+        return {};
+      ifOp = opIf;
+      continue;
+    }
+    if (!isVerilogExpression(&op))
```

Comment:
don't you also need to check emitter.outOfLineExpressions to ensure the expression isn't marked for out-of-line emission?

Reply:
Doesn't matter -  everything is hoisted to the scope containing the outermost if, as indicated by the NestedElseIfHost test case.